go/api/base_client.go

```diff
@@ -662,6 +662,39 @@ func (client *baseClient) SUnion(keys []string) (map[Result[string]]struct{}, er
 	return handleStringSetResponse(result)
 }
 
+func (client *baseClient) SScan(key string, cursor string) (Result[string], Result[[]string], error) {
```

Comment:
`Result[[]string]` means that either `[]string` is present or is `nil`. To represent this information, `[]string` need not be wrapped inside `Result` as a slice can be returned as `nil`. The ideal thing would be to wrap `string` inside `Result` and return `[]Result[string]`.

I hope the intention of wrapping `int64`, `bool`, `string`, `float64` inside `Result` is clear. Any other types like `slice`, `map` are not required to be wrapped inside `Result`. Please let me know if there are any questions.

Reply:
I see. So as far as I understand the Result type is introduced so that we can always have nullability in types, and we don't want to be command-specific when it comes to return types, I suppose we are gonna have to wrap the string elements in Results for this command even though they don't seem to be nullable in the server reply. Is that correct?